modules/Sources/UIComponents/Text/RestoringWalletBadge.swift

```diff
@@ -0,0 +1,88 @@
+//
+//  RestoringWalletBadge.swift
+//  
+//
+//  Created by Lukáš Korba on 18.12.2023.
+//
+
+import SwiftUI
+import Generated
+
+public struct RestoringWalletBadgeModifier: ViewModifier {
+    public enum Background {
+        case pattern
+        case solid
+        case transparent
+    }
+    
+    let isOn: Bool
+    let background: Background
+
+    public func body(content: Content) -> some View {
+        if isOn {
+            ZStack(alignment: .top) {
+                content
+                    .zIndex(0)
+                
+                if background == .pattern {
```

Comment:
This could reduce code duplication by setting a variable to the argument of `.background` conditionally, and then the rest of the code (`RestoringWalletBadge()...`) is the same.

Reply:
The situation is more complicated here. First branch sets the `image` in the background while the else branch sets the `color`. SwiftUI won't compile because I didn't provide opaque type and SwiftUI can't decide it because it's one or other. That's why I decided to "duplicate" these 3 lines

```swift
.frame(maxWidth: .infinity)
.padding(.bottom, 6)
.zIndex(1)
```

and let the code be readable and compact. There are some possible solutions to remove the duplicity but the cost is always some other layer of complexity. This way it's "the most clean" SwiftUI way.